src/Umbraco.Tests/Scoping/ScopeUnitTests.cs

```diff
@@ -72,6 +74,31 @@ public void WriteLock_Acquired_Only_Once_Per_Key()
             syntaxProviderMock.Verify(x => x.WriteLock(It.IsAny<IDatabase>(), Constants.Locks.Languages), Times.Once);
         }
 
+        [Test]
+        public void WriteLock_Acquired_Only_Once_When_InnerScope_Disposed()
+        {
+            var scopeProvider = GetScopeProvider(out var syntaxProviderMock);
+
+            using (var outerScope = scopeProvider.CreateScope())
+            {
+                outerScope.WriteLock(Constants.Locks.Languages);
+
+                using (var innerScope = scopeProvider.CreateScope())
+                {
+                    innerScope.WriteLock(Constants.Locks.Languages);
+                    innerScope.WriteLock(Constants.Locks.ContentTree);
+                    innerScope.Complete();
+                }
+
+                outerScope.WriteLock(Constants.Locks.ContentTree);
+                outerScope.Complete();
+            }
+
+            // Since we request the ReadLock after the innerScope has been dispose, the key has been removed from the dictionary, and we fetch it again.
```

Comment:
I don't understand this comment?

Reply:
Ah, that was a note for myself of why the test initially failed when I wrote it, I just forgot to delete it and it's no longer relevant I'll remove it 😄